metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceImpl.java

```diff
@@ -109,13 +115,19 @@ public void notifyOfPartitionAddition(final MetacatSaveTablePartitionPostEvent e
         final long timestamp = event.getRequestContext().getTimestamp();
         final String requestId = event.getRequestContext().getId();
         // Publish a global message stating how many partitions were updated for the table to the table topic
+        final TablePartitionsUpdatePayload partitionsUpdatePayload;
+        if (this.config.isSnsNotificationAttachPartitionIdsEnabled() && event.getPartitions() != null) {
```

Comment:
Move the if logic inside the create method

Reply:
ya, we can move the config check inside, but not the other one since we separated the events as MetacatDeleteTablePartitionPostEvent and MetacatSaveTablePartitionPostEvent, don't want to check the class then cast inside.